less_slow.cpp

```diff
@@ -1814,6 +1843,39 @@ BENCHMARK_CAPTURE(theoretic_tops, i7_amx_avx512, tops_i7_amx_avx512fma_asm_kerne
 
 #pragma region Alignment of Memory Accesses
 
+/**
+ *  For the inline expansion compiler optimization, different
+ *  compilers use different attributes (more modern) or
+ *  keywords (in older implementations, or for compatibility).
+ *
+ *  It is common to see a cross-platform/compiler definition
+ *  that covers the most common ones used in libraries.
+ */
+
+#if defined(_MSC_VER)
+#define LESS_SLOW_ALWAYS_INLINE [[msvc::forceinline]] inline
+#elif defined(__GNUC__)
+#define LESS_SLOW_ALWAYS_INLINE [[gnu::always_inline]] inline
+#elif defined(__clang__)
+#define LESS_SLOW_ALWAYS_INLINE [[clang::always_inline]] inline
+#else
+#define LESS_SLOW_ALWAYS_INLINE inline __attribute__((always_inline))
+#endif
+
+/**
+ *  @brief  Checks if a number is a power of two.
+ *
+ *  TODO: use intrinsics when available; possibly use std::popcount;
+ *  benchmark between them?
+ */
```

Comment:
This is an awesome place to link Sean Eron Anderson's [Bit Twiddling Hacks](https://graphics.stanford.edu/~seander/bithacks.html) and Hacker's Delight by Henry S. Warren Jr. 🤗

Reply:
Great reference, I'll add that in